From bd152bec1086a35f426c53c9bb9f0c309e216037 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Wed, 2 Nov 2011 19:07:10 +0100
Subject: [PATCH 5/5] am335x: Change mmc_load_uimage to load at a fixed address
 The kernel normally expects to be run from 0x80008000 so if we
 load from SD card at that-mkimage header we can save a copy of
 the kernel.

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
---
 include/configs/am335x_evm.h |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/configs/am335x_evm.h b/include/configs/am335x_evm.h
index 56e36da..8c0fae2 100755
--- a/include/configs/am335x_evm.h
+++ b/include/configs/am335x_evm.h
@@ -74,8 +74,8 @@
 	"loadbootenv=fatload mmc ${mmc_dev} ${loadaddr} ${bootenv}\0" \
 	"importbootenv=echo Importing environment from mmc ...; " \
 		"env import -t $loadaddr $filesize\0" \
-	"mmc_load_uimage=fatload mmc ${mmc_dev} ${loadaddr} ${bootfile}\0" \
-	"mmc_load_uimage_ext2=ext2load ${mmc_dev} ${loadaddr} /boot/${bootfile}\0" \
+	"mmc_load_uimage=fatload mmc ${mmc_dev} 0x80007fc0 ${bootfile}\0" \
+	"mmc_load_uimage_ext2=ext2load ${mmc_dev} 0x80007fc0 /boot/${bootfile}\0" \
 	"optargs=\0" \
 	"bootargs_defaults=setenv bootargs " \
 		"console=${console} " \
@@ -103,7 +103,7 @@
 		"ip=dhcp\0" \
 	"mmc_boot=run mmc_args; " \
 		"run mmc_load_uimage; " \
-		"bootm ${loadaddr}\0" \
+		"bootm 0x80007fc0\0" \
 	"nand_boot=echo Booting from nand ...; " \
 		"run nand_args; " \
 		"nand read.i ${loadaddr} ${nand_src_addr} ${nand_img_siz}; " \
@@ -137,7 +137,7 @@
 		"fi;" \
 		"if run mmc_load_uimage; then " \
 			"run mmc_args;" \
-			"bootm ${loadaddr};" \
+			"bootm 0x80007fc0;" \
 		"fi;" \
 	"fi;" \
 	"run nand_boot;" \
-- 
1.7.2.5

